go/master/etcd_store.go

```diff
@@ -0,0 +1,133 @@
+package master
+
+import (
+	"context"
+	"sync"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/concurrency"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// DefaultLockPath is the default etcd master lock path.
+	DefaultLockPath = "/master/lock"
+	// DefaultStatePath is the default etcd key for master state.
+	DefaultStatePath = "/master/state"
+)
+
+// EtcdStore is the Store implementation backed by etcd.
+type EtcdStore struct {
+	lockPath  string
+	statePath string
+	ttlSec    int
+	client    *clientv3.Client
+
+	mu   sync.Mutex
+	lock *concurrency.Mutex
+}
+
+// NewEtcdStore creates a new EtcdStore.
+func NewEtcdStore(endpoints []string, lockPath, statePath string, ttlSec int) (*EtcdStore, error) {
+	cli, err := clientv3.New(clientv3.Config{
```

Comment:
I really don't know when to do `cli.Close()` to release etcd client resources, shall we catch signal `INT` and then call a `DeleteEtcdStore`? Or we can leave this to OS in this version.

I've also created an issue to talk about "how to gracefully shutdown master and pservers"